sw/airborne/modules/ctrl/ctrl_windtunnel.c

```diff
@@ -0,0 +1,180 @@
+#define WINDTUNNEL_TO_BODY_PSI 0
+#endif
+
+struct ctrl_windtunnel_struct {
+  int rc_throttle;
+  int rc_roll;
+  int rc_pitch;
+  int rc_yaw;
+  struct OrientationReps rotation;
+} ctrl_windtunnel;
+
+float ctrl_windtunnel_steptime = 8.0;
+struct min_max_ctrl_t ctrl_windtunnel_throttle = {.min = 0, .max = 0, .step = 1200};
+struct min_max_ctrl_t ctrl_windtunnel_flaps = {.min = -7200, .max = 0, .step = 2400};
+static float last_time = 0;
+
+void ctrl_module_init(void);
+void ctrl_module_run(bool in_flight);
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+static void send_windtunnel_meas(struct transport_tx *trans, struct link_device *dev)
+{
+  struct FloatQuat windtunnel_to_body;
+  struct FloatEulers windtunnel_to_body_e;
+  struct FloatQuat *rotation = orientationGetQuat_f(&ctrl_windtunnel.rotation);
+  struct FloatQuat *ned_to_body = stateGetNedToBodyQuat_f();
+  float_quat_comp_inv(&windtunnel_to_body, ned_to_body, rotation);
+  float_eulers_of_quat(&windtunnel_to_body_e, &windtunnel_to_body);
+
+  float aoa = DegOfRad(windtunnel_to_body_e.theta);
+  float power = electrical.vsupply * electrical.current;
+  pprz_msg_send_WINDTUNNEL_MEAS(trans, dev, AC_ID, &aoa, &air_data.airspeed, &electrical.vsupply, &electrical.current, &power, COMMANDS_NB, stabilization_cmd);
+}
+#endif
+
+void ctrl_module_init(void)
+{
+  ctrl_windtunnel.rc_throttle = 0;
+  ctrl_windtunnel.rc_roll = 0;
+  ctrl_windtunnel.rc_pitch = 0;
+  ctrl_windtunnel.rc_yaw = 0;
+
+  // Create a rotation to the windtunnel from the body
+  struct FloatEulers windtunnel_to_body_eulers =
+  {WINDTUNNEL_TO_BODY_PHI, WINDTUNNEL_TO_BODY_THETA, WINDTUNNEL_TO_BODY_PSI};
+  orientationSetEulers_f(&ctrl_windtunnel.rotation, &windtunnel_to_body_eulers);
+
+#if PERIODIC_TELEMETRY
+  register_periodic_telemetry(DefaultPeriodic, PPRZ_MSG_ID_WINDTUNNEL_MEAS, send_windtunnel_meas);
+#endif
+}
+
+void ctrl_module_run(bool in_flight __attribute__((unused)))
+{
+  bool done = false;
+  // Increase step in steptime
+  if(ctrl_windtunnel.rc_throttle > (MAX_PPRZ/2) && (get_sys_time_float() - last_time) > ctrl_windtunnel_steptime) { 
+    // Increase throttle step if flaps at the end
+    if((ctrl_windtunnel_flaps.current + ctrl_windtunnel_flaps.step) > ctrl_windtunnel_flaps.max) {
+      // Only increase step if throttle is not at the end
+      if((ctrl_windtunnel_throttle.current + ctrl_windtunnel_throttle.step) <= ctrl_windtunnel_throttle.max) {
+        ctrl_windtunnel_throttle.current += ctrl_windtunnel_throttle.step;
+        ctrl_windtunnel_flaps.current = ctrl_windtunnel_flaps.min;
+        last_time = get_sys_time_float();
+      }
+      else {
+        // Finished
+        done = true;
+      }
+    }
+    else {
+      // By default increase flaps
+      ctrl_windtunnel_flaps.current += ctrl_windtunnel_flaps.step;
+
+      // Double the amount of steptime during double transition
+      if((ctrl_windtunnel_flaps.current == ctrl_windtunnel_flaps.min))
```

Comment:
avoid equality comparison between float

Reply:
in this case, the ```min``` value is assigned to ```current``` by design so the comparison works.